stdlib/LinearAlgebra/src/bidiag.jl

```diff
@@ -239,9 +239,9 @@ for func in (:conj, :copy, :real, :imag)
     @eval ($func)(M::Bidiagonal) = Bidiagonal(($func)(M.dv), ($func)(M.ev), M.uplo)
 end
 
-adjoint(B::Bidiagonal) = Adjoint(B)
-transpose(B::Bidiagonal) = Transpose(B)
-adjoint(B::Bidiagonal{<:Real}) = Bidiagonal(B.dv, B.ev, B.uplo == 'U' ? :L : :U)
+adjoint(  B::Bidiagonal) = Bidiagonal(  adjoint.(B.dv),   adjoint.(B.ev), B.uplo == 'U' ? :L : :U)
+transpose(B::Bidiagonal) = Bidiagonal(transpose.(B.dv), transpose.(B.ev), B.uplo == 'U' ? :L : :U)
+adjoint(B::Bidiagonal{<:Real})     = Bidiagonal(B.dv, B.ev, B.uplo == 'U' ? :L : :U)
```

Comment:
If memory serves, the existing implementation was a conscious decision. The reason that comes to mind immediately is that it avoids unnecessary allocations, but something in the back of my mind says there might've been a correctness issue as well.

If memory serves, we went with the existing implementation to avoid unnecessary allocation.

Reply:
I can't really see why there'd be a correctness issue here but you are right about the allocations since it basically makes the operations eager. I'll revert this part and overload the relevant `*` and `\` methods for `Adjoint{T,Bidiagonal}` etc instead.

Reply:
Sounds good! And agreed, I have difficulty seeing where a correctness issue would exist as well; chances are that memory is for something else related to `Bidiagonal` and `Adjoint`/`Transpose`. Best!

Reply:
(Please feel welcome to request review if you would like another round after revisions by the by. Best!)